Assets/Scenes/Scripts/GameManager.cs

```diff
@@ -15,34 +15,51 @@ public class GameManager : MonoBehaviour {
     void PlayerManager() {
         player.transform.Translate(new Vector3(Input.GetAxis("Horizontal"), Input.GetAxis("Vertical"), 0) * Time.deltaTime * playerMoveSpeed);
         projectileSpawnTime += Time.deltaTime;
-        if (projectileSpawnTime >= 1f / playerMainClassLevel) { // level determines how fast the player can shoot, TODO: extract this for multiple classes
+        if (projectileSpawnTime >= 1f / playerMainClassLevel) {
             GameObject projectile = Instantiate(projectilePrefab, player.transform.position + new Vector3(1, 0, 0), Quaternion.identity);
             projectiles.Add(projectile);
             StartCoroutine(SpiralAndFade(projectile));
             projectileSpawnTime = 0;
         }
+        if (player.GetComponent<Collider2D>().IsTouchingLayers(LayerMask.GetMask("Enemy"))) {
+            playerHealth -= enemyAttackDamage; // Handle player damage (Example : play sound, show effect)
+        }
     }
     private IEnumerator SpiralAndFade(GameObject obj) { // Spriral and fade effect for projectiles #1 animation for projectiles
         float angle = 0, startTime = Time.time;
         Renderer objRenderer = obj.GetComponent<Renderer>();
         Color originalColor = objRenderer.material.color;
-        while (Time.time - startTime < fadeDuration) {
+        while (Time.time - startTime < fadeDuration && obj != null) {
             angle += spiralSpeed * Time.deltaTime;
             obj.transform.position = player.transform.position + new Vector3(Mathf.Cos(angle), Mathf.Sin(angle), 0) * Mathf.Min(Time.time - startTime, 2.0f);
             objRenderer.material.color = new Color(originalColor.r, originalColor.g, originalColor.b, Mathf.Lerp(1, 0, (Time.time - startTime) / fadeDuration));
             yield return null;
         }
-        Destroy(obj);
+        if (obj != null) {
+            Destroy(obj);
+        }
     }
     void EnemyManager() {
         enemySpawnTime += Time.deltaTime;
         if (enemySpawnTime >= enemySpawnInterval) {
             enemies.Add(Instantiate(enemyPrefab, player.transform.position + (Vector3)Random.insideUnitCircle.normalized * 10f, Quaternion.identity));
             enemySpawnTime = 0;
         }
-        foreach (var enemy in enemies) {
-            if (enemy != null) {
+        List<GameObject> enemiesCopy = new List<GameObject>(enemies);
+        foreach (var enemy in enemiesCopy) {
+            if (enemy != null){
                 enemy.GetComponent<Rigidbody2D>().MovePosition(enemy.GetComponent<Rigidbody2D>().position + (Vector2)(player.transform.position - enemy.transform.position).normalized * Time.deltaTime * enemyMoveSpeed);
+                foreach (var projectile in projectiles) {
+                    if (projectile != null && projectile.GetComponent<Collider2D>().IsTouching(enemy.GetComponent<Collider2D>())) {
+                        enemyHealth -= projectileDamage;
+                        Destroy(projectile);
```

Comment:
**Discussion:** What you think, let's answer on this review for any idea?

**My idea:** Maybe not destroy when it's touch to enemy, lets it destroy them self so it means it can deal damage to multiple enemy until they out of energy in this case lets it fade away and destroy

Reply:
- The Projectile can deal damage to enemies until it runs out of energy or reaches the end of its use time
- Maybe can use the Projectile Damage Reduction by implement where each collision reduces the projectiles energy Once the energy reaches zero the projectiles gets destroyed or the Projectiles will gets destroyed after deal damage to a certain number of enemies

Reply:
Great!, I like your idea, it feels this make our 100 line more ลูกเล่น. Do it then. 555555

Reply:
Also, can you fix the bug that enemies get destroyed by one projectile? Maybe it is because the health of each is not separate from the other.

Reply:
commit separate feat: projectile ....., and fix: enemies ....

Reply:
need random health for each enemy ?

Reply:
No, curenntly enemy health are share for each other

Enemy health start at 50 it will take 3 projectile to die but after take first 3 projectile all of them have negative health so it takes only 1 projectile then dies.

(currently no matter which enemy take damage it all share to others)
**1st enemy 50 hp**
take **damage -20** (hit only 1 enemy)
1st enemy **30hp,** **other enemy also 30hp**
**another enemy** that **not the same one take damage -20**
1st enemy 10hp (WTF!)

Reply:
Ok now 
which one ?
- The Projectile can deal damage to enemies until it runs out of energy or reaches the end of it use time  ?
or
- the Projectile will get destroyed after deal damage to a certain number of enemies ?

Reply:
The Projectile can deal damage to enemies until it runs out of energy or reaches the end of it use time

Reply:
เอาเป็น use time นะ | around 2-3 sec ?

Reply:
จริงๆ คือมันมีอยู่แล้วนะ แค่เอา destroy บรรทัด 55 ออก
โค้ดที่มันสร้างลูกกลมๆ (บรรทัด 21 เรียกใช้ 28) มันจะหายไปเองตามค่าของ fadeDuration ตั้งไว้สัก ~4.5, spiralSpeed ~1-2 มันจะหมุนรอบตัว player พอดี

Reply:
โอเช